src/crimson/osd/pg_backend.cc

```diff
@@ -178,6 +178,70 @@ static inline bool _read_verify_data(
   return true;
 }
 
+repair_errorator::future<> PGBackend::rep_repair_primary_object(
+  const hobject_t& soid,
+  OSDOp& osd_op) 
+{
+  /*
+  TODO: 
+  1. Create and maintain the data structure `waiting_for_unreadable_object`
```

Comment:
instead of maintaining a dedicated `waiting_for_unreadable_object`, we use 

https://github.com/ceph/ceph/blob/9745a4cd1999ab2f36042b93e7ade6fa76fc8eda/src/crimson/osd/osd_operations/client_request.cc#L128-L131

to wait until a given object is recovered.

Reply:
What is the difference between UrgentRecovery and: https://github.com/ceph/ceph/blob/e5f917aa1c2f6c473ece6667b108c325f0151da5/src/crimson/osd/pg.cc#L236-L246
Is the latter needed if we use the former? Notice in my commit that I am starting a LocalPeeringEvent (as a replacement to `queue_peering_event()` in classical-osd)

Reply:
discussed with Amnon offline. will only address the corrupted case using `UrgentRecovery` machinery in this change.